### ISSUE TITLE
Is the data missing?

### ISSUE THREAD

**Issue by cuizhuzhuchen** [zd11024/NaviLLM#23]

**Comment by zd11024** [comment]
For the first question, please refer to #11 .
For the second question, it seems that there is a syntax error in your bash scripts? Maybe you can check the script more carefully. If you have any questions, please let me know.

**Comment by cuizhuzhuchen** [comment]
Thank you very much for your previous answer. Can I modify some parameters to perform inference on 8 NVIDIA 4090 (24G, and these 8 cards can only perform remote computing without interface display)? For example, I changed batch_size to 1, but it still shows that the video memory is insufficient, see the picture for details. Are there any other methods?
![8](https://github.com/user-attachments/assets/ac266e55-6492-4f12-95dc-799b9cfbf4e7)
![9](https://github.com/user-attachments/assets/f7223cec-8438-483d-9a68-004cbdabccb7)

**Comment by zd11024** [comment]
1. One simple way is to decrease the navigation step in the config `configs/multi.yaml` as shown in the following figure. However, this will reduce performance during evaluation. 
![image](https://github.com/user-attachments/assets/5aa15f61-6502-4ae6-8d56-63a7043b2eed)

2. Another way is to modify the data parallelism to the naive model parallelism, which will increase the inference time. In this case, you have to change the function `dist_models` in `tools/optims.py` and the inference script.

**Comment by cuizhuzhuchen** [comment]
Thank you very much for your answer, but the two methods you provided still don't work. After studying the code, I found that there are a lot of codes, the connections between them are complicated, and various errors often occur. Is there any explanation about the code here? In addition, can the 8 4090s really be trained and inferred?

**Comment by zd11024** [comment]
Sorry, the issue does not show your latest response, but I received your reply in my email.
You can also replace the Vicuna-7B in the model with a smaller model. We have retained the interface for OPT in `models/modifed_lm.py`, but we did not save the corresponding checkpoint. You can also utilize other smaller model.
In addition, You could also use LoRA for parameter-efficient finetuning.

**Comment by cuizhuzhuchen** [comment]
![image](https://github.com/user-attachments/assets/59874f4e-be97-4f74-a98b-781f31c2eff8)
![image](https://github.com/user-attachments/assets/d3d30dde-d03e-4a1f-b13d-8dd9887df7db)
![image](https://github.com/user-attachments/assets/030f2511-85bf-47f5-bdc6-50ad71ebd8d3)
Thank you very much for your answer. I will try the method mentioned above. For the previous question, I changed the parameter nproc_per_node from 8 to 1, and I can complete the inference on 4090. The result is a json file, but I have not figured out the content of the file and how the result corresponds to the data in the paper. I tested the command sh scripts/multi_wo_pretrain.sh, and it still requires 80G of video memory.

**Comment by cuizhuzhuchen** [comment]
![image](https://github.com/user-attachments/assets/797a89ad-2138-4287-89a4-79e0398774f2)
![image](https://github.com/user-attachments/assets/5bc3a2b2-b364-47b7-98cc-eac421c631f5)
![image](https://github.com/user-attachments/assets/55724c46-53ab-45c1-bdc5-4bbd3d8d8522)
Hello, I would like to ask how to calculate the results in the paper based on the results I inferred? Take CVDN (the test subset
consists of 1,384 questions over 16 environments) as an example. The second picture above has the results after the test. Based on the results, how can I quickly calculate the GP under TEST in the first picture as 7.9? And in the json file obtained from the test, how can I calculate the Test values ​​of SOON, R2R, REVERIE, and ScanQA in Figure 1 as 26.26, 60, 32.33, and 26.3.The third picture is the result of the SOON test.

**Comment by zd11024** [comment]
For these benchmarks, you need to submit the json file to eval.ai to calculate the corresponding results.

**Comment by cuizhuzhuchen** [comment]
Thank you very much for your answers, CVDN, SOON, REVERIE,ScanQA, we have obtained the results, in which challenge should R2R be tested?

**Comment by cuizhuzhuchen** [comment]
Thank you very much for your answer. CVDN, SOON, REVERIE, ScanQA, we have obtained the results. In which challenge should R2R be tested? Since eval.ai has a limited number of times, how can I manually test these results?

**Comment by zd11024** [comment]
R2R is tested at this [link](https://eval.ai/web/challenges/challenge-page/97/overview). Since the test set is not open to public, you can only submit it to the test server for evaluation.

**Comment by zd11024** [comment]
I am closing this issue for now. Please feel free to reopen it if needed.
